src/utils/workflow_utils/src/workflow_utils.c

```diff
@@ -1868,6 +1868,7 @@ char* workflow_get_root_sandbox_dir(const ADUC_WorkflowHandle handle)
         }
 
         tempRet = pwf;
+        pwf = NULL;
         Log_Debug("Using parent workfolder: '%s'", pwf);
```

Comment:
Log should happen before transfer ownership, or use tempRet instead of pwf here.